Core/include/Acts/TrackFitting/KalmanFitter.hpp

```diff
@@ -289,7 +289,7 @@ class KalmanFitter {
     const Surface* targetSurface = nullptr;
 
     /// Allows retrieving measurements for a surface
-    const std::map<GeometryIdentifier, SourceLink>* inputMeasurements = nullptr;
+    std::map<GeometryIdentifier, SourceLink> inputMeasurements;
```

Comment:
Are you sure this doesn't cause extra copies?

Reply:
I think it does but the pointer is also weird. but we can `move` if you prefer